### ISSUE TITLE
Не видны сообщения в skype, отправленные ботом

### ISSUE THREAD

**Issue by bopoda** [denisyfrolov/azure-functions-php-bot#1]
Спасибо за статью на habr`е!
Я впервые создаю бота и пользуюсь Azure.

Бота создал https://dev.botframework.com/bots на портале, добавил его контакт в свой desktop skype.
Сообщения, который я посылаю боту, видны на portal.azure.com в службах приложений в моём приложении.

Но, вот, ответы от бота не приходят в мой desktop skype. Хотя, как понимаю, в коде присутствует обратная отправка для recipient?

Может, вы сталкивались и могли бы подсказать с чем может быть связано?)

**Comment by denisyfrolov** [comment]
Да, проблему вижу. Microsoft Bot Connector API стал возвращать 401 на попытку отправить сообщение пользователю с токеном из login.microsoftonline.com. Судя по всему, с тех пор как я писал статью, изменилась процедура авторизации бота. Если сможете исправить, кидайте пуллреквест, если нет.. тогда придется подождать, пока у меня появится время свободное.

**Comment by bopoda** [comment]
@denisyfrolov спасибо.
Если будет время и возможность - создам.
Если вы сделаете раньше - отлично.

**Comment by ovikeee** [comment]
@bopoda Я тоже долго пытался исправить данную ошибку, в итоге порывшысь в доках, выяснил, что с  July 31, 2017 изменился scope. 
https://docs.microsoft.com/en-us/azure/bot-service/rest-api/bot-framework-rest-connector-authentication 
`Support for v3.0 of the security protocol was discontinued on July 31, 2017. If you have written your own authentication code (i.e., did not use the Bot Builder SDK to create your bot), you must upgrade to v3.1 of the security protocol by updating your application to use the v3.1 values that are listed below.
`     
замените следующий код:
           array(
                    'grant_type' => 'client_credentials',
                    'client_id' => $client_id, //ID приложения
                    'client_secret' => $client_secret, //Пароль приложения
                    'scope' => **'https://api.botframework.com/.default'** //'https://graph.microsoft.com/.default'								
                )
